agent/src/agentclient.rs

```diff
@@ -358,6 +393,10 @@ impl<T: APIServerClient> BrupopAgent<T> {
                     // TODO: we need add some criterias here by which we decide to transition
                     // from MonitoringUpdate to WaitingForUpdate.
                 }
+                BottlerocketShadowState::ErrorReset => {
+                    // Spec should never be ErrorReset
+                    panic!("Spec state should never be ErrorReset");
+                }
```

Comment:
I might not understand what you meant. Do you mean something similar to 
```
    let token_path = token_path.to_str().context(error::Assertion {
        message: "Token path (defined in models/agent.rs) is not valid unicode.",
    })?;
```
But this just gives out an error to let the caller handle. In this case, we want the program to panic since this should not happen. The difference would just be panic here or the caller panic later.

Reply:
Yeah. I think the idea is to have the caller handle gracefully terminating the program, since our outer error handler is writing to the termination log it will be properly logged. If we let the program panic here, it will be really difficult to determine why it failed.

Reply:
Do you know if there is better way to write this?
```
agentclient_error::Error::Assertion { message: msg } => {
                            return agentclient_error::Assertion { message: msg }.fail();
                        }
```